client/me/purchases/membership-item/index.tsx

```diff
@@ -1,69 +1,123 @@
-						amount: formatCurrency( Number( subscription.renewal_price ), subscription.currency ),
-					},
-				} ) }
-			</div>
-		);
+		return <>{ translate( 'Never expires' ) }</>;
 	}
 
-	/* Renews every month for $5. Next renewal on November 22, 2020. */
-	if ( subscription.renew_interval ) {
+	return (
+		<>
+			{ translate( 'Renews at %(amount)s on %(date)s', {
+				args: {
+					amount: formatCurrency( Number( subscription.renewal_price ), subscription.currency ),
+					date: moment( subscription.end_date ).format( 'LL' ),
+				},
+			} ) }
+		</>
+	);
+};
+
+const SiteLink = ( { subscription }: { subscription: MembershipSubscription } ) => {
+	const translate = useTranslate();
+
+	return (
+		<button
+			className="membership-item__site-name purchase-item__link"
+			onClick={ ( event ) => {
+				event.stopPropagation();
+				event.preventDefault();
+				window.location = subscription.site_url;
+			} }
+			title={ translate( 'Visit %(siteName)s', {
+				args: {
+					siteName: subscription.site_title,
+				},
+			} ) }
+		>
+			{ subscription.site_title }
+		</button>
+	);
+};
+
+const MembershipType = ( { subscription }: { subscription: MembershipSubscription } ) => {
+	const translate = useTranslate();
+
+	if ( subscription.end_date === null ) {
 		return (
-			<div className="membership-item__term-label">
-				{ translate( 'Renews every %(interval)s for %(amount)s. Next renewal on %(date)s.', {
-					args: {
-						interval: subscription.renew_interval,
-						amount: formatCurrency( Number( subscription.renewal_price ), subscription.currency ),
-						date: moment( subscription.end_date ).format( 'LL' ),
+			<>
+				{ translate( 'Purchased from {{site}}{{/site}}', {
+					components: {
+						site: <SiteLink subscription={ subscription } />,
 					},
 				} ) }
-			</div>
+			</>
 		);
 	}
 
-	/* Renews at $5 on November 22, 2020. */
-	/* I'm not sure we can have a renewal without an interval, so this might not get called. */
 	return (
-		<div className="membership-item__term-label">
-			{ translate( 'Renews at %(amount)s on %(date)s.', {
-				args: {
-					amount: formatCurrency( Number( subscription.renewal_price ), subscription.currency ),
-					date: moment( subscription.end_date ).format( 'LL' ),
+		<>
+			{ translate( 'Subscription to {{site}}{{/site}}', {
+				components: {
+					site: <SiteLink subscription={ subscription } />,
 				},
 			} ) }
-		</div>
+		</>
 	);
 };
 
+const Icon = ( { subscription }: { subscription: MembershipSubscription } ) => {
+	const [ hasError, setErrors ] = useState( false );
+	const [ site, setSite ] = useState( null );
+	const [ loadData, setLoadData ] = useState( true );
+
+	async function fetchData() {
+		const data = await fetch(
+			'https://public-api.wordpress.com/rest/v1.1/sites/' + subscription.site_id
+		);
+
+		data
+			.json()
+			.then( ( data ) => {
+				setSite( data );
+				setLoadData( false );
+			} )
+			.catch( ( err ) => setErrors( err ) );
+	}
+
+	useEffect( () => {
```

Comment:
This will re-fetch the data repeatedly on every render until one of them succeeds. That might not be what you want. It's usually a good idea to provide a dependency array to `useEffect`.

Reply:
There was something weird happening and that's why I added the condition to check if the data had loaded here: https://github.com/Automattic/wp-calypso/pull/47610/files/468995d0e7e0f204b4ad8081aadc2e46b2029293#diff-f84151fb3113ed3ece0d28671295958b9b8f443b388cceef48763749aa6ee3f8R109. 

Can you explain what you mean by providing a dependency array?

Reply:
Yeah, the condition check is a good idea! Sorry I didn't mention this earlier!

`useEffect` with one (function) argument says, "after render, run this function".

`useEffect` can have a second argument, which is an array of variables which it calls "dependencies". If the dependency array is present, `useEffect` means, "after render, run this function, but only if the identity of one of these variables has changed". If the array is empty, it means to run only once.

So `useEffect(() => fetchData())` means "fetch data on every render". `useEffect(() => fetchData(), [isLoaded])` means "fetch data on every render that `isLoaded` has changed". `useEffect(() => fetchData(), [])` means "fetch data once when the component first renders".

So this `useEffect` here says "fetch data on every render, but don't fetch if we have completed fetching". This is a little tricky because since fetching the data is async, there could be any number of renders before `loadData` becomes false. It will re-fetch the data during each of those renders. 

Notice also that if there's an error, `loadData` remains true, so it will continue fetching on every render.

Reply:
It's not a big bug, I just wanted to mention it as it should be cleaned up eventually. Worst case the browser will be fetching that data a lot, but everything should still work.

Reply:
Cool, yeah no problem. Happy to clean it up. Thanks for the background!

Reply:
No worries. Nice job with this refactor, by the way! 👏

Reply:
Thank you! I got fix posted here: https://github.com/Automattic/wp-calypso/pull/48242. Let me know if that's better.